src/seq_join.rs

```diff
@@ -117,14 +117,55 @@ where
     seq_join(active, iterable).try_collect()
 }
 
+enum ActiveItem<F: IntoFuture> {
+    Pending(Pin<Box<F::IntoFuture>>),
+    Resolved(F::Output),
+}
+
+impl<F: IntoFuture> ActiveItem<F> {
+    /// Drives this item to resolved state when value is ready to be taken out. Has no effect
+    /// if the value is ready.
+    ///
+    /// ## Panics
+    /// Panics if this item is completed
+    fn check_ready(&mut self, cx: &mut Context<'_>) -> bool {
+        let mut ready = true;
+        match self {
+            ActiveItem::Pending(f) => {
+                if let Poll::Ready(v) = Future::poll(Pin::as_mut(f), cx) {
+                    *self = ActiveItem::Resolved(v);
+                } else {
+                    ready = false;
+                }
+            }
+            ActiveItem::Resolved(_) => {}
+        }
+
+        ready
+    }
+
+    /// Takes the resolved value out
+    ///
+    /// ## Panics
+    /// If the value is not ready yet.
+    #[must_use]
+    fn take(self) -> F::Output {
+        if let ActiveItem::Resolved(v) = self {
+            v
+        } else {
+            panic!("No value to take out")
+        }
```

Comment:
Is this better?  I don't know any more.

```suggestion
        let ActiveItem::Resolved(v) = self else {
            panic!("No value to take out");
        };
        v
```

Reply:
4 < 5 so it is strictly better